src/modules/authentification/components/AuthLoginForm.tsx

```diff
@@ -1,48 +1,138 @@
-import { useForm, FormProvider } from "react-hook-form";
-import InputField from "./inputs/InputField";
+import { auth, provider } from "../../firebase/firebase";
+import {
+  signInWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
 import {
   passwordMaxLength,
   passwordMinLength,
   passwordRegexp,
+  emailRegexp,
 } from "../const/userInputRequirements";
+import { userState, loggedState } from "../state/userState";
+import { useSetRecoilState } from "recoil";
+import { useNavigate, Link } from "react-router-dom";
+import googleLogoImg from "../assets/google-logo.png";
+import InputField from "./inputs/InputField";
+import { useForm, FormProvider } from "react-hook-form";
+import Button from "../../../shared/Button";
+
+interface LoginData {
+  email: string;
+  password: string;
+}
 
 export default function AuthLoginForm() {
-  const form = useForm();
+  const navigate = useNavigate();
+
+  const form = useForm<LoginData>();
```

Comment:
You can make a custom hook, lets say useAuthLogin, where you can put all the logic you are using in this component, and kave a cleaner code.

So you would have:
`const {form, handleGoogleSignIn } = useAuthLofin()`

Reply:
Added useAuthLogin and useAuthRegister hooks for login and register logic.